lib/at_cmd/at_cmd.c

```diff
@@ -109,25 +119,82 @@ static int get_return_code(char *buf, struct return_state_object *ret)
 	return new_len;
 }
 
-static void callback_worker(struct k_work *item)
+static inline int at_write(const char *const cmd)
 {
-	struct callback_work_item *data =
-		CONTAINER_OF(item, struct callback_work_item, work);
+	int bytes_sent;
+	int bytes_to_send = strlen(cmd);
+
+	LOG_DBG("Sending command %s", log_strdup(cmd));
 
-	if (data != NULL) {
-		data->callback(data->data);
+	bytes_sent = send(common_socket_fd, cmd, bytes_to_send, 0);
+
+	if (bytes_sent == -1) {
+		LOG_ERR("Failed to send AT command (err:%d)", errno);
+		return -errno;
 	}
 
-	k_mem_slab_free(&rsp_work_items, (void **)&data);
+	if (bytes_sent != bytes_to_send) {
+		LOG_WRN("Bytes sent (%d) was not the same as expected (%d)",
+			bytes_sent, bytes_to_send);
+	}
+
+	return 0;
+}
+
+/* Clear the current command safely */
+static void complete_cmd(void)
+{
+	k_mutex_lock(&current_cmd_mutex, K_FOREVER);
```

Comment:
Isn't this mutext already locked at this point when comming from line 186.

Reply:
Yes, it is doubly locked in that case